src/main/java/com/zaxxer/hikari/HikariConfig.java

```diff
@@ -1018,6 +1042,18 @@ private void validateNumerics()
          maxLifetime = MAX_LIFETIME;
       }
 
+      // keepalive time must larger then 30 seconds
+      if (keepaliveTime != 0 && keepaliveTime < SECONDS.toMillis(30)) {
+         LOGGER.warn("{} - keepaliveTime is less than 30000ms, disabling it.", poolName);
+         keepaliveTime = DEFAULT_KEEPALIVE_TIME;
+      }
+
+      // keepalive time must be less than maxLifetime (if maxLifetime is enabled)
+      if (keepaliveTime != 0 && maxLifetime != 0 && keepaliveTime < maxLifetime) {
+         LOGGER.warn("{} - keepaliveTime is less than maxLifetime, disabling it.", poolName);
+         keepaliveTime = DEFAULT_KEEPALIVE_TIME;
+      }
```

Comment:
Sorry, I really don't know what I was thinking. I shouldn't review code at 4am. Of course it is valid for ``keepaliveTime`` and ``maxLifetime`` to *both* be set. In that case, of course, ``keepaliveTime`` *should* be less than ``maxLifetime``. So, the check should be *inverted*:
```java
if (keepaliveTime != 0 && maxLifetime != 0 && keepaliveTime >= maxLifetime) {
   LOGGER.warn("{} - keepaliveTime is greater than or equal to maxLifetime, disabling it.", poolName);
   keepaliveTime = DEFAULT_KEEPALIVE_TIME;
}
```

Reply:
That's my fault. I didn't review your code but just copy it.